lib/encryption.rb

```diff
@@ -0,0 +1,41 @@
+class Encryption
+  attr_reader :character_set
+
+  def initialize
+    @character_set = ("a".."z").to_a << " "
+  end
+
+  def indexed_alphabet
+    index_to_letter = @character_set.group_by do |letter|
+      @character_set.index(letter)
+    end
+
+    index_to_letter.transform_values! do |value|
+      value.join
+    end
+  end
+
+  def alphabet_to_index
+    indexed_alphabet.invert
+  end
+
+  def encrypt_message(message, shifts)
+    encrypted_message = ""
+
+    message.downcase.each_char.with_index do |character, i|
+      if @character_set.include?(character)
+        key = shifts[i % shifts.length]
+        alphabet_index = alphabet_to_index[character]
+
+        if (alphabet_index + key) < 27
+          encrypted_message << indexed_alphabet[alphabet_index + key]
+        else
+          encrypted_message << indexed_alphabet[(alphabet_index + key) % 27]
```

Comment:
MODULO <3 

Used here to ensure that no matter how big the shift is, it will always find the right index of the shifted character. Started off with subtracting 27, but that only works up until a shift of 54, and then it fails. Modulo returns the remainder which works for all whole integer values!